test/test_contract.js

```diff
@@ -122,98 +131,714 @@ describe('Chess contract', function() {
+          }, Error);
+
+          // white
+          Chess.setGameState(gameId, state, player1, {from: player1, gas: 2000000});
+          assert.throws(function() {
+            Chess.move(gameId, 116, 118, {from: player1, gas: 500000});
+          }, Error);
+          assert.throws(function() {
+            Chess.move(gameId, 116, 114, {from: player1, gas: 500000});
+          }, Error);
+        });
+
+        it('should reject castling when check 2', () => {
+          let state = [...defaultBoard];
+          // clear black side
+          state[1] = 0;
+          state[2] = 0;
+          state[3] = 0;
+          state[5] = 0;
+          state[6] = 0;
+          // clean white side
+          state[113] = 0;
+          state[114] = 0;
+          state[115] = 0;
+          state[117] = 0;
+          state[118] = 0;
+
+          // clear pawns
+          state[21] = 0;
+          state[99] = 0;
+
+          state[55] = 3; // W_B
+          state[48] = -3; // B_B
+
+          // black
+          Chess.setGameState(gameId, state, player2, {from: player1, gas: 2000000});
+          assert.throws(function() {
+            Chess.move(gameId, 4, 6, {from: player2, gas: 500000});
+          }, Error);
+          assert.throws(function() {
+            Chess.move(gameId, 4, 2, {from: player2, gas: 500000});
+          }, Error);
+
+          // white
+          Chess.setGameState(gameId, state, player1, {from: player1, gas: 2000000});
+          assert.throws(function() {
+            Chess.move(gameId, 116, 118, {from: player1, gas: 500000});
+          }, Error);
+          assert.throws(function() {
+            Chess.move(gameId, 116, 114, {from: player1, gas: 500000});
+          }, Error);
+        });
+
+        it('should reject later en passant', () => {
+          let state = [...defaultBoard];
+          state[100] = 0;
+          state[52] = 1; // W_P
+
+          Chess.setGameState(gameId, state, player2, {from: player1, gas: 2000000});
+          assert.doesNotThrow(() => {
+            Chess.move(gameId, 19, 51, {from: player2, gas: 500000});
+          });
+          assert.doesNotThrow(() => {
+            Chess.move(gameId, 96, 64, {from: player1, gas: 500000});
+            Chess.move(gameId, 21, 53, {from: player2, gas: 500000});
+          });
+          assert.throws(() => {
+            Chess.move(gameId, 52, 35, {from: player1, gas: 500000});
+          }, Error);
+        });
+
+        it('should reject invalid en passant', () => {
+          assert.doesNotThrow(() => {
+            Chess.move(gameId, 100, 68, {from: player1, gas: 500000});
+            Chess.move(gameId, 19, 51, {from: player2, gas: 500000});
+
+            Chess.move(gameId, 68, 52, {from: player1, gas: 500000});
+          });
+          assert.throws(() => {
+            Chess.move(gameId, 51, 68, {from: player2, gas: 500000});
+          }, Error);
+        });
+
+        it('should reject pawn promotion to king or pawn');
```

Comment:
If we want to implement another promotion this test has to be there. If we do not want this, this can be removed.